uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/API/Wrapper/LeaveRequestDatesTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestDates as LeaveRequestDates;
+
+/**
+ * Class CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestDates
+ *
+ * @group headless
+ */
+class CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestDatesTest extends BaseHeadlessTest {
+
+  /**
+   * @var \CRM_HRLeaveAndAbsences_API_Wrapper_LeaveRequestDates
+   */
+  private $wrapper;
+
+  public function setUp() {
+    $this->wrapper = new LeaveRequestDates();
+  }
+
+  public function testRequestParamsAreNotUpdatedIfTheRequestIsNotForLeaveRequestGetOrGetFull() {
+    $apiRequest = [
+      'entity' => 'LeaveRequest',
+      //some random action name, so that we have different actions for different tests
+      'action' => 'get' . rand(10, 1000),
+      'params' => [
+        'test' => '123'
+      ]
+    ];
+
+    $wrappedRequest = $this->wrapper->fromApiInput($apiRequest);
+
+    $this->assertEquals($apiRequest, $wrappedRequest);
```

Comment:
Would the `$apiRequest` be different even if the `canHandleTheRequest` check passed?

Reply:
Good point. I guess I can change the params to include the fields handled by the wrapper to make this a little bit better